### PR TITLE
Delete previous comments

### PR DESCRIPTION
**Pull request by kasty**
Hi,

Here's a new proposal : delete previous comments before adding a new one. It can be disabled with a new `delete-comment` input (default is set to `true`)

Thanks again for your excellent base, it was very easy to tweak !

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -49,18 +49,41 @@ ${summary}
...
+
+  return Promise.all(
+    data
+      .filter(
+        (c) =>
+          c.user && c.body && c.user.login === 'github-actions[bot]' && c.body.indexOf('Total Coverage') !== -1
+      )
+      .map((c) => octokit.issues.deleteComment({ ...context.repo, comment_id: c.id }))
+  );
+};
```
Comment (raulanatol) [inline src/utils.ts]: I think that this method is difficult to read. I think that can be more readable similar to this:

```js
    const { data: comments } = await octokit.issues.listComments({
        ...context.repo,
        per_page: 100,
        issue_number: issueNumber
    });
    return Promise.all(
        comments.filter(isPreviousTotalCoverageComment)
            .map(deleteComment(octokit)));
```

```js
const isPreviousTotalCoverageComment = comment => {
    comment.user &&
    comment.body &&
    comment.user.login === 'github-actions[bot]' &&
    comment.body.includes('Total Coverage');
};

const deleteComment = octokit => comment =>
    octokit.issues.deleteComment({ ...context.repo, comment_id: comment.id });
```

File: src/utils.ts [hunk 2]
```diff
@@ -49,18 +49,41 @@ ${summary}
...
 export const createComment = async (comment: string) => {
   const octokit = getOctokit(getInput('github-token'));
   const issueNumber = getIssueNumber(context.payload);
+  const deletePrev = getBooleanInput('delete-previous');
   if (!issueNumber) {
     warning('Issue number not found. Impossible to create a comment');
     return;
   }
 
+  if(deletePrev) {
```
Comment (raulanatol) [inline src/utils.ts]:
```suggestion
  if (deletePrev) {
```

😝

**Comment by raulanatol** [comment]
Good idea @kasty 👏🏼 👏🏼

File: src/utils.ts (hunk 1 above)
Comment (kasty) [reply src/utils.ts]: I agree with you ! I'll update this :)

**Review by raulanatol — APPROVED** [review]

**Pull request merged**